File: angular/src/shared/Dto/user-declaration-model.ts
export class UserDeclarationModel implements IUserDeclarationModel {
   
    Id: number | undefined
    UserId: number | undefined;
    CompanyName: string | undefined;
    ContractTypeId: number | undefined;
    Interest: string | undefined;
    ApprovalStatusId: number | undefined;
    StartYear: number | undefined;
    EndYear: number| undefined;
    ApprovedBy : number| undefined;    
    CreatedBy: number| undefined;
    constructor(data?: UserDeclarationModel) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }
    code: string;
    name: string;
    id: number;

    init(data?: any) {
        if (data) {
            this.Id = data["Id"];
            this.UserId = data["UserId"];
            this.CompanyName = data["CompanyName"];
            this.ContractTypeId = data["ContractTypeId"];
            this.Interest = data["Interest"];
            this.ApprovalStatusId = data["ApprovalStatusId"];
            this.StartYear = data["StartYear"];
            this.EndYear = data["EndYear"];
            this.ApprovedBy = data["ApprovedBy"];
            this.CreatedBy = data["CreatedBy"];
            
        }
    }

    static fromJS(data: any): UserDeclarationModel {
        data = typeof data === 'object' ? data : {};
        let result = new UserDeclarationModel();
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["Id"] = this.Id;
        data["UserId"] = this.UserId;
        data["CompanyName"] = this.CompanyName;
        data["ContractTypeId"] = this.ContractTypeId;
        data["Interest"] = this.Interest;
        data["ApprovalStatusId"] = this.ApprovalStatusId;
        data["StartYear"] = this.StartYear;
        data["EndYear"] = this.EndYear;
        data["ApprovedBy"] = this.ApprovedBy;
        data["CreatedBy"] = this.CreatedBy;
        return data; 
    }

    clone(): UserDeclarationModel {
        const json = this.toJSON();
        let result = new UserDeclarationModel();
        result.init(json);
        return result;
    }
}


export interface IUserDeclarationModel {
    Id: number | undefined
    UserId: number | undefined;
    CompanyName: string | undefined;
    ContractTypeId: number | undefined;
    Interest: string | undefined;
    ApprovalStatusId: number | undefined;
    StartYear: number | undefined;
    EndYear: number| undefined;
    ApprovedBy : number| undefined;
    CreatedBy: number| undefined;

    
   
}

File: Ozone.WebApi/Ozone.Application/DTOs/Security/UserDeclarationModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Ozone.Application.DTOs
{
   public class UserDeclarationModel
    {
        public long Id { get; set; }
        public long? UserId { get; set; }

        public string CompanyName { get; set; }
        public long? ContractTypeId { get; set; }
        public string ContractTypeName { get; set; }

        public string Interest { get; set; }
        public long? ApprovalStatusId { get; set; }
        public string ApprovalStatus { get; set; }
        public int? StartYear { get; set; } 
     
        public int? EndYear { get; set; }
        public bool? IsActive { get; set; }
        public bool? IsDeleted { get; set; }
        public long? ApprovedBy { get; set; }

        public DateTime? ApprovedDate { get; set; }
        public long? CreatedBy { get; set; }

        public DateTime? CreatedDate { get; set; }
    }
}
